### ISSUE TITLE
Rule to prevent declaration of class fields as arrow functions

### ISSUE THREAD

**Issue by mkay581** [eslint/eslint#13747]
**Please describe what the rule should do:**

Prevent declaring arrow function declarations on class fields. This is not allowed in ES6. 

```js
class A { 
    foo = () => {
        console.log('foo from A')
    }
}

```

So it would be nice to have a rule that prevents contributors from doing it. The code above should be ES-compliant and changed to:

```js
class A { 
    foo() {
        console.log('foo from A')
    }
}

```

**What new ECMAScript feature does this rule relate to?**

https://github.com/tc39/proposal-class-fields

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
class A { 
    foo = () => {
        console.log('foo from A')
    }
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

Declaring methods in this way are not supported by [Class Fields](https://github.com/tc39/proposal-class-fields) proposal for EcmaScript so it makes sense to place this rule in ESLint proper.

**Are you willing to submit a pull request to implement this rule?**

Perhaps if I'm able to carve out some time to understand how to implement.

**Comment by mdjermanovic** [comment]
Hi @mkay581, thanks for the new rule proposal!

The Class Fields proposal is still in stage 3.

Per our [policy about experimental features](https://github.com/eslint/eslint#what-about-experimental-features), ESLint's default parser and ESLint core rules support stage 4 features only. This also applies to new rules that would disallow some stage 3 features, since the rule and the parser should then have to be aware of the still-experimental syntax.

I'm closing this as it isn't actionable at the moment. Please feel free to re-submit the proposal if and when this syntax advances to stage 4, and we'd be happy to discuss it.
